firmware/application/ui_navigation.cpp

```diff
@@ -1009,6 +1010,86 @@ void BMPView::paint(Painter&) {
         portapack::display.drawBMP({(240 - 230) / 2, (320 - 50) / 2 - 10}, splash_bmp, true);
 }
 
+bool BMPView::on_touch(const TouchEvent event) {
+    /*
+     * TODO: since the touch screen detect touching evt policy is kind of bad, with hard-coded threshold it works but not sensitive, i will use single touch firstly
+     * and it's a waist of space to implement them all and as options in settings.
+     * the touch screen policy can be better, talked here https://discord.com/channels/719669764804444213/956561375155589192/1198926225897443328
+     * when that fixed (so the threshold won't make the corner always been touched), just un-comment one of the field and two var in header to make it works.
+     */
+
+    if (!nav_.is_valid()) {
+        return false;
+    }
+
+    switch (event.type) {
+        case TouchEvent::Type::Start:
+            chThdSleep(10);
+            handle_pop();
+            return false;
+
+        case TouchEvent::Type::End:
```

Comment:
this is because when in a bad screen it always pressing a corner, and the end event is necessary in this case cuz the start event isn't working correctly.